metrics/Gauge.md

```diff
@@ -0,0 +1,80 @@
+# Gauge API Overview
+A `Gauge` is used to record already aggregated metrics, or metrics that can go up and down. Typical examples of gauges would be the number of jobs/entries in a queue or number of threads in a running state.
+
+The `Gauge` values can be negative. This document describes the key types and the overall bahavior of API.
+
+## Gauge API
+
+The value that is published for gauges is an instantaneous measurement of an `int64` or `double` value. This API is useful when you want to manually increase and decrease values as per service requirements.
+
+The following general operations MUST be provided by the API:
+* Defining a `name`, `description`, `unit`, `labelKeys` and `constantLabels` which are fixed labels that always apply to a gauge. This should give back the gauge object to getOrcreate time series, remove time series and clear all time series.
+	* `name`: a string describing the name of the metric, e.g. "vm_cpu_cycles" or "queue_size". Names MUST be unique within the library. It is recommended to use names compatible with the intended end usage.
+	* `description`: a string describing the metric, e.g."Virtual cycles executed on VM".
+	* `unit`: a string describing the unit used for the metric. Follows the format described by
+[Unified Code for Units of Measure](http://unitsofmeasure.org/ucum.html).
+	* `labelKeys`: the list of the label keys to track different types of metric.
+	* `constantLabels`: the map of label keys and label values.
+* Add a new time series with label values, which returns a `Point` (which is part of the `TimeSeries`). Each point represents an instantaneous measurement of a varying gauge value. Each Gauge Metric has one or more time series for a single metric.
+	* `labelValues`: the list of label values. The number of label values must be the same to that of the label keys.
+* The `Point` class should provide functionalities to manually increment/decrement values. Example: `add(long amt)`, `set(long value)`.
+* Remove a single time series from the gauge metric, if it is present.
+	* `labelValues`: the list of label values.
+* Clear all time series from the gauge metric i.e. References to all previous point objects are invalid (not part of the metric).
+* Get a default time series for a gauge with all labels not set, or default labels.
+
+Example in Java:
+```java
+private static final MetricRegistry metricRegistry = Metrics.getMetricRegistry();
+
+List<LabelKey> labelKeys = Arrays.asList(LabelKey.create("key1", "description"));
+List<LabelValue> labelValues = Arrays.asList(LabelValue.create("queue1"));
+
+LongGauge gauge = metricRegistry.addLongGauge("queue_size", "The number of jobs", "1", labelKeys);
```

Comment:
Constant labels?

Reply:
I think this should be the optional param, right?

Reply:
It's optional but good to show the usage in the examples.

Reply:
Maybe we should use a Builder in java because the list of options becomes long enough. Also there are optional things like unit that can be default "1"

Reply:
Done, PTAL